### ISSUE TITLE
Figure 2 show gradient

### ISSUE THREAD

**Issue by fengbeiliu** [shengliu66/ELR#10]
Hi,

Do you have a script to reproduce Figure 2? I am just curious how to get gradient about different loss function part. Could you provide some pointers on this?

Thanks

**Comment by shengliu66** [comment]
Hi,

If you are using pyTorch, there is a function called register_backward_hook() to save gradients for variables.

Best,
Sheng
